### PR TITLE
Add curl dependency to opentelemetry-cpp elasticsearch feature

### PR DESCRIPTION
**Pull request by aristotelos**
If `curl` is not listed as a dependency, the port build fails.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by aristotelos** [comment]
@microsoft-github-policy-service agree

File: ports/opentelemetry-cpp/vcpkg.json [hunk 1]
```diff
@@ -23,7 +23,10 @@
   ],
   "features": {
     "elasticsearch": {
-      "description": "Whether to include the Elasticsearch Client in the SDK"
+      "description": "Whether to include the Elasticsearch Client in the SDK",
+      "dependencies": [
+        "curl"
```
Comment (dg0yt) [inline ports/opentelemetry-cpp/vcpkg.json:28]: Don't block users' opt-out from default features:
```suggestion
        {
          "name": "curl",
          "default-features": false
        }
```

Comment (aristotelos) [reply ports/opentelemetry-cpp/vcpkg.json]: OK, pushed an update

**Comment by aristotelos** [comment]
@jimwang118 I see you added the label "port-feature". Isn't this a bug in the port (rather than a feature) because it does not specify required dependencies?

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**